geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala

```diff
@@ -54,21 +58,24 @@ class AttributeIdxStrategy(val filter: QueryFilter) extends Strategy with LazyLo
       (intervals.map(_._1.getMillis).min, intervals.map(_._2.getMillis).max)
     }
 
-    // for an attribute query, the primary filters are considered an OR
-    // (an AND would never match unless the attribute is a list...)
-    val propsAndRanges = {
-      val primary = filter.singlePrimary.getOrElse {
-        throw new IllegalStateException("Attribute index does not support Filter.INCLUDE")
-      }
-      getBounds(sft, primary, dates)
-    }
-    val attribute = propsAndRanges.headOption.map(_.attribute).getOrElse {
-      throw new IllegalStateException(s"No ranges found for query filter $filter. " +
-          "This should have been checked during query planning")
+    // TODO currently exclusive ANDs will return nothing - but if it's a list type it could actually match
```

Comment:
actually, it's more far-reaching than I thought at first - I don't think we've ever supported this - we'd need to apply the filter to the results coming back if it's a list. I'll open a ticket.